test/Dialect/Calyx/errors.mlir

```diff
@@ -1050,3 +1050,99 @@ module attributes {calyx.entrypoint = "main"} {
     }
   }
 }
+
+// ----- 
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}) {
+    %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op the input for 'calyx.invoke' is empty.}}
+      calyx.invoke@r() -> ()
+    }
+  }
+}
+
+// ----- 
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}) {
+    %c10 = hw.constant 10 : i32
+    %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1 
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op has a cell port as the destination with the incorrect direction.}}
+      calyx.invoke@r(%r.out = %c10) -> (i32)
+    }
+  }
+}
+
+// ----- 
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}) {
+    %c1 = hw.constant 1 : i1
+    %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1 
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op the go port of 'r' cannot appear here.}}
```

Comment:
`r` does not have a `go` port so it's not clear what this error message is referring to